### DISCUSSION TITLE
Serial/Sequential ID properties for Types

### DISCUSSION THREAD

**Discussion by eisenhaus335** [ArcadeData/arcadedb#1729] [Ideas]
#1723 A bit relevance
I have a need for serial sequential ID for tables. Although, I understand the performance bottlenecks when using sequential ID creation but for my own use case. My usecase will be minimal, just for invoicing unique sequential ID. I think Serial/Sequential default type properties will be useful for those who have a need for simple integer ID. Arcadedb engine could store latest id number so that user not need to make query returning the max ID `SELECT max(sequenceid) as latest_id FROM Invoice` whenever they need to make vertex. It just be nice if arcadedb engine provide them, with precaution expected and reduced throughput write-power

If this idea are not received. Maybe, we could try suggest for those who have kind of this use. For example, You need to use sqlscript (which is transaction) and unique index so that you don't cause ID duplication. The alternative kind is making their own document which store latest sequential ID which stores the latest ID (don't forget using sqlscript!)

I refer good articles for those who need it
https://www.cybertec-postgresql.com/en/postgresql-sequences-vs-invoice-numbers/

On unrelated notes and with my vague understanding. I think arcadedb engine are fine to not solve distributed database computing problem. This is kinda related when you need to generate some kind of distributed business data problem. I think the db is not for that. We already have relational database so it might be for them. Arcadedb engine may serve limited use case but for single instance complex multi-role database with high performing throughput write. I mean, even for Neo4j struggles on distributed scale computing problem haha, i think. But, then it maybe will be next problem for enterprise-alike business data. I am fine using arcadedb for knowledge base for small-scale complex-data business type of usecase

**Comment by mdre** [comment]
We need sequentials too like ones OrientDB has.
